app/src/main/java/org/mozilla/fenix/perf/RunBlockingCounter.kt

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunBlockingCheck")
+
+package org.mozilla.fenix.perf
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+object RunBlockingCounter {
+    var count = AtomicInteger(0)
+
+    fun incrementCounter() {
+        var prev: Int
+        var next: Int
+        do {
+            prev = count.get()
+            next = prev + 1
+            if (next == Integer.MAX_VALUE) next = 0
```

Comment:
nit: while we're not going to hit this case, I think it's more honest to cap it at `Integer.MAX_VALUE` rather than reset the counter. e.g. if you look at the counter after running the program after a few days and its 100, you might be a little confused.